### PR TITLE
Task/status bar notification

### PR DESCRIPTION
**Pull request by dimacodota**

### CODE REVIEW

**Comment by dimacodota** [comment]
nee to add test

File: src/extension.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ import pollNotifications, {
   cancelNotificationsPolling,
 } from "./notifications/pollNotifications";
 import handleAlpha from "./alphaInstaller";
+import pollMessages, { dispose } from "./statusBarMessages/pollStatusBarMessages";
```
Comment (boaz-codota) [inline src/extension.ts:25]: It is called a `status bar` so why call the thing `message` and not `status`? It is a bar of statuses...

File: src/extension.ts [hunk 2]
```diff
@@ -70,6 +72,7 @@ async function backgroundInit(context: vscode.ExtensionContext) {
 export async function deactivate(): Promise<unknown> {
   void closeValidator();
   cancelNotificationsPolling();
+  dispose();
```
Comment (boaz-codota) [inline src/extension.ts:75]: Give it a more meaningful name, such as `disposeStatus`

File: src/notifications/pollNotifications.ts [hunk 3]
```diff
@@ -44,7 +45,11 @@ async function handleNotification(
   context: vscode.ExtensionContext
 ): Promise<void> {
   try {
-    await assertFirstTimeRecieved(notification.id, context);
+    await assertFirstTimeReceived(notification.id, context);
+
+    void setState({
+      [StatePayload.MESSAGE]: { message: notification.message, message_type: StateType.NOTIFICATION },
+    });
```
Comment (boaz-codota) [inline src/notifications/pollNotifications.ts:52]: I'd love to know what that does

File: src/consts.ts [hunk 4]
```diff
@@ -14,6 +14,20 @@ export const DELAY_FOR_CODE_ACTION_PROVIDER = 800;
 export const BINARY_NOTIFICATION_POLLING_INTERVAL = +(
   process.env.BINARY_NOTIFICATION_POLLING_INTERVAL || 10_000
 ); // 10 seconds
+
+export const BINARY_STATUS_BAR_POLLING_INTERVAL = +(
+  process.env.BINARY_STATUS_BAR_POLLING_INTERVAL || 3600000
```
Comment (boaz-codota) [inline src/consts.ts:19]: prefer multiplications for readability: `3600000` -> `60 * 60 * 1_000`

File: src/consts.ts [hunk 5]
```diff
@@ -14,6 +14,20 @@ export const DELAY_FOR_CODE_ACTION_PROVIDER = 800;
...
+); // one hour
+
+export const BINARY_STATUS_BAR_FIRST_MESSAGE_POLLING_INTERVAL = +(
+  process.env.BINARY_NOTIFICATION_POLLING_INTERVAL || 10_000
+); // 10 seconds
+
+export const STATUS_BAR_NOTIFICATION_PERIOD = +(
+  process.env.STATUS_BAR_NOTIFICATION_PERIOD ||  60000 * 2
```
Comment (boaz-codota) [inline src/consts.ts:27]: `60000 * 2` -> `2 * 60 * 1_000`

File: src/statusBar.ts [hunk 6]
```diff
@@ -40,6 +47,12 @@ export function setErrorStatus(issue?: string | undefined | null): void {
   statusBar.color = new ThemeColor("errorForeground");
 }
 
+export function setPromotionStatus(message: string, command: string): void{
+  statusBar.text = `${ATTRIBUTION_BRAND}${BRAND_NAME} : ${message}`;
```
Comment (boaz-codota) [inline src/statusBar.ts:51]: there should be no space between `${BRAND_NAME}` and `:`

File: src/statusBarMessages/pollStatusBarMessages.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+export function dispose(): void {
+  if(statusBarCommandDisposable){
+    statusBarCommandDisposable.dispose();
+  }
+  cancelFirstMessagePolling();
+  cancelHourlyMessagePolling();
+  resetToDefaultStatus();
+}
```
Comment (boaz-codota) [inline src/statusBarMessages/pollStatusBarMessages.ts:105]: file is over 100 lines - split it

File: src/binary/requests/statusBar.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
+
+import { tabNineProcess } from "./requests";
+
+export enum StatusBarActionActions {
+  NONE = 0,
+  OPEN_LP = 1,
```
Comment (boaz-codota) [inline src/binary/requests/statusBar.ts:6]: I did a mistake when I did this thing in the notifications, and we don't need to repeat it. The binary is opening the LP so this air conditioning placement is redundant and will bite us in the future.

Comment (boaz-codota) [reply src/binary/requests/statusBar.ts:6]: In the minimum, just omit the OPEN_LP option so it will be clear that nothing is done

**Review by boaz-codota — CHANGES_REQUESTED** [review]

**Review by boaz-codota — APPROVED** [review]
🚀

**Pull request merged**